### DISCUSSION TITLE
Securing api routes in NextJs

### DISCUSSION THREAD

**Discussion by billtlee** [Web3Auth/Web3Auth#492] [Help]
Hi,
We are using web3auth to authenticate users on our NextJs app.  It works well on the client side.  What is the best approach to secure api routes on the server?  Thanks in advance!

Bill

**Comment by himanshuchawla009** [comment]
You can use the idToken returned by web3auth to secure you backend api endpoints, plz refer to this doc to know more:-
https://web3auth.io/docs/server-side-verification/

**Reply by shahbaz17** [reply]
Here's an example:

https://github.com/Web3Auth/examples/blob/main/web-core-sdk/server-side-verification/ssv-via-social-nextjs-core-example/pages/App.tsx#L93

https://github.com/Web3Auth/examples/blob/main/web-core-sdk/server-side-verification/ssv-via-social-nextjs-core-example/pages/api/login.ts

**Reply by billtlee** [reply]
Thanks!
